syntax = "proto3";

import "google/protobuf/timestamp.proto";

import "protos/yagpcc_metrics.proto";
import "protos/yagpcc_plan.proto";

package yagpcc;
option java_outer_classname = "SegmentYAGPCCAS";
option go_package = "a.yandex-team.ru/cloud/mdb/yagpcc/api/proto/agent_segment;greenplum";

message SetQueryReq {
    QueryStatus           query_status = 1;
    google.protobuf.Timestamp datetime = 2;
    QueryKey                 query_key = 3;
    SegmentKey             segment_key = 4;
    QueryInfo               query_info = 5;
    GPMetrics            query_metrics = 6;
    repeated MetricPlan      plan_tree = 7;
}

message SetPlanNodeReq {
    PlanNodeStatus         node_status = 1;
    google.protobuf.Timestamp datetime = 2;
    QueryKey                 query_key = 3;
    SegmentKey             segment_key = 4;
    GPMetrics             node_metrics = 5;
    MetricPlan               plan_node = 6;
}
